Pcap++/header/PcapFilter.h

```diff
@@ -484,79 +484,105 @@ namespace pcpp
 
 
 	/**
-	 * @class AndFilter
-	 * A class for connecting several filters into one filter with logical "and" between them. For example: if the 2 filters are: "IPv4 address =
-	 * x.x.x.x" + "TCP port dst = 80", then the new filter will be: "IPv4 address = x.x.x.x _AND_ TCP port dst = 80"<BR>
-	 * This class follows the composite design pattern<BR>
+	 * @class CompositeFilter
+	 * A class for connecting several filters into one filter using a delimiter between them. For example: if the 2 filters are: "IPv4 address =
+	 * x.x.x.x" + "TCP port dst = 80", then the new filter will be: "IPv4 address = x.x.x.x _DELIMITER_ TCP port dst = 80"<BR>
+	 * The primary use for this filter is to provide base functionality for composite filters such as AndFilter and OrFilter.<BR>
 	 * For deeper understanding of the filter concept please refer to PcapFilter.h
-	 * @todo add some methods: "addFilter", "removeFilter", "clearAllFilter"
 	 */
-	class AndFilter : public GeneralFilter
+	class CompositeFilter : public GeneralFilter
 	{
 	private:
+		std::string m_CompositeDelimiter;
 		std::vector<GeneralFilter*> m_FilterList;
 	public:
-
 		/**
-		 * An empty constructor for this class. Use addFilter() to add filters to the and condition
+		 * A constructor that sets the composite delimiter. Use addFilter() to add filters to the composite filter.
+		 * @param[in] compositeDelimiter A delimiter to add between different filters.
 		 */
-		AndFilter() {}
+		explicit CompositeFilter(std::string compositeDelimiter) : m_CompositeDelimiter(std::move(compositeDelimiter)){};
 
 		/**
-		 * A constructor that gets a list of pointers to filters and creates one filter from all filters with logical "and" between them
+		 * A constructor that gets a list of pointers to filters and creates one filter from all filters
 		 * @param[in] filters The list of pointers to filters
+		 * @param[in] compositeDelimiter A delimiter to add between different filters.
 		 */
-		explicit AndFilter(std::vector<GeneralFilter*>& filters);
+		explicit CompositeFilter(std::vector<GeneralFilter*>& filters, std::string compositeDelimiter);
```

Comment:
let's make it to `const string&`

Reply:
The change to `const string&` will prevent parsing string literals to being passed to the constructor, which is the way AndFilter and OrFilter specify their delimiter. The change will also require either adding a r-value reference constructor being added or wrapping the string literal into a `std::string` before passing it to the constructor as a const reference.
Which solution would you prefer?

Reply:
r-value reference constructor makes sense.

Reply:
Resolved in e3be18e

It ended up not requiring a r-value constructor to pass c-strings, so its just the `string` -> `const string&` change.